kroxylicious/src/test/java/io/kroxylicious/proxy/internal/FilterHandlerTest.java

```diff
@@ -79,6 +86,34 @@ public void testForwardResponse() {
         assertEquals(frame, propagated, "Expect it to be the frame that was sent");
     }
 
+    @Test
+    public void testOtherFiltersInChainCanFilterOutOfBandResponse() {
+        ApiVersionsResponseFilter recipientFilter = taggingApiVersionsResponseFilter("recipient");
+        String filterName = "other-interested-filter";
+        ApiVersionsResponseFilter filterUnderTest = taggingApiVersionsResponseFilter(filterName);
+        buildChannel(filterUnderTest);
+        CompletableFuture<Object> future = new CompletableFuture<>();
+        var frame = writeInternalResponse(new ApiVersionsResponseData(), future, recipientFilter);
+        var propagated = channel.readInbound();
+        assertEquals(frame, propagated, "Expect it to be the frame that was sent");
+        assertResponseMessageTaggedWith(filterName, (InternalResponseFrame<?>) propagated);
+        assertFalse(future.isDone());
+    }
+
+    @Test
+    public void testOtherFiltersInChainCanFilterOutOfBandRequest() {
+        ApiVersionsRequestFilter recipientFilter = taggingApiVersionsRequestFilter("recipient");
```

Comment:
I haven't quite groked where you are going with this test.  I was expected to see a filter make the call to io.kroxylicious.proxy.internal.DefaultFilterContext#sendRequest and then a verification made that the request and its response goes through the filter.

Reply:
Ah, I was working at the level the FilterHarness offers currently, which checks how the FilterHandler behaves for just one frame and filter.

So these tests are checking, "if the frame is out-of-band and the recipient is not this filter, this filter can intercept and forward the message".

Hope that makes some sense.